apps/web/src/pages/profile.tsx

```diff
@@ -34,59 +133,166 @@ const ProfilePage: NextPageWithLayout = () => {
-        className="flex py-1 font-medium"
-        onClick={() => router.back()}
-      >
-        <ArrowLeft />
-        <span>Go Back</span>
-      </DashboardButton>
-      <div className="mx-auto mt-12 max-w-xl">
-        <h1 className="text-3xl font-bold">Account Settings</h1>
-        <form onSubmit={onSubmit} className="my-8 space-y-8">
-          <Avatar
-            role="img"
-            imageUrl={data?.image ?? ""}
-            userName={data?.name ?? data?.email ?? ""}
-            className="mx-auto block h-36 w-36"
-          />
-          <div>
-            <label htmlFor="name" className="block text-sm font-medium">
-              Name *
-            </label>
-            <div className="mt-1">
-              <Input
-                type="text"
-                name="name"
-                id="name"
-                defaultValue={data?.name ?? ""}
+    <>
+      <div>
+        <DashboardButton
+          className="flex py-1 font-medium"
+          onClick={() => router.back()}
+        >
+          <ArrowLeft />
+          <span>Go Back</span>
+        </DashboardButton>
+        <div className="mx-auto mt-12 max-w-3xl space-y-3">
+          <section className="rounded-xl bg-gray-800 px-6 py-3">
+            <h1 className="text-3xl font-bold">Account Settings</h1>
+            <form onSubmit={onSubmit} className="mt-8 space-y-8">
+              <Avatar
+                role="img"
+                imageUrl={data?.image ?? ""}
+                userName={data?.name ?? data?.email ?? ""}
+                className="mx-auto block h-36 w-36"
               />
+              <div>
+                <label htmlFor="name" className="block text-sm font-medium">
+                  Name *
+                </label>
+                <div className="mt-1">
+                  <Input
+                    type="text"
+                    name="name"
+                    id="name"
+                    defaultValue={data?.name ?? ""}
+                  />
+                </div>
+              </div>
+              <div>
+                <label htmlFor="name" className="block text-sm font-medium">
+                  Email *
+                </label>
+                <div className="mt-1">
+                  <Input
+                    type="text"
+                    defaultValue={data?.email ?? ""}
+                    disabled
+                  />
+                </div>
+              </div>
+              <div>
+                You signed up in via{" "}
+                <span className="font-medium capitalize underline underline-offset-2">
+                  {data?.accounts[0]?.provider}
+                </span>
+              </div>
+              <DashboardButton
+                type="submit"
+                className="ml-auto block px-5 py-1"
+              >
+                Save
+              </DashboardButton>
+            </form>
+          </section>
+          <section className="rounded-xl bg-gray-800 px-6 py-3">
```

Comment:
maybe its time for dedicated components?

Reply:
yup